### ISSUE TITLE
hook-use-state false positive detection of destructured value

### ISSUE THREAD

**Issue by rajzik** [yannickcr/eslint-plugin-react#3252]
I have typescript with react and I am using `useState` hook with destructured values.

`example.ts`

```ts
function useHook(props: { initialValue?: number, minValue?: number }) {
  const { initialValue, minValue } = props; 
  const [internalValue, setValue] = useState<number | '-' | ''>(initialValue ?? minValue ?? 1);
}
```

## Result

<img width="485" alt="image" src="https://user-images.githubusercontent.com/10364836/159271327-7a0ce0ac-f5b6-4927-a371-099e9806c7a6.png">

## Reproduction repository

https://github.com/rajzik/eslint-react-repro

## Eslint settings

https://github.com/Oriflame/lumos/blob/63c032f2bd74a8aababf14b31fb2fee4b66b498e/packages/eslint-config/src/rules/react.ts#L23

**Comment by rajzik** [comment]
@ljharb wouldn't be possible to change hint? I didn't get it from current wording.

**Comment by ljharb** [comment]
Absolutely it would - a PR with your suggested wording would be appreciated.
